### DISCUSSION TITLE
How to extract frames as bitmap from webp

### DISCUSSION THREAD

**Discussion by KishorJena** [coil-kt/coil#1216] [Q&A]
Hi, please help me to decode animated webp frame by frame. I want to resize each bitmaps and later encode them into webp image file. This is not for drawing on image views. I just want to get file from local and save it to local storage.
I tried ImageMagick CLI it was working fine but it's android binding does not cover all the functionality of imagemgaick so I had to remove it. ffmpeg does not have animated decoder. After googling and exploring forums I came to here,I read that coil has animated webp decoder.  At least give a pseudo code that it is possible in coil so that I will start implementing.

**Comment by rishabhhgupta** [comment]
so late i got [this solution ](https://github.com/b4rtaz/android-webp-encoder/blob/master/app/src/main/java/com/n4no/webpencoder/webp/graphics/WebpBitmapEncoder.java)

**Comment by rishabhhgupta** [comment]
it's only have encoder no decoder, all i need is to get all the bimap arraylist of webp, each frame rate and total Duration, only solution i think i have is use webplib which is written in c/c++ that i have to import in my android studio and i didn't have experience of c/c++, as you say you create decoder but i didn't find , how i decode the animated webp file

**Reply by rishabhhgupta** [reply]
> Yeah I stopped coding related to webp encoder decoder. Now i remember that the PR was just to implement the alpha channel to preserve the transparency of animated webp images. and later published the decoder encoder on my github as I have made major changes in encoder as well and wrote decoder codes. [Webp_Transcoder](https://github.com/KishorJena/Webp_Transcoder) usage:
> 
> ```
>        WebpAnimatedDecoder deocder = new WebpAnimatedDecoder(new File(inFile));
>        List<ProcessedFrames> processedFrames = deocder.decode();
> 
>        // collect all frames as bitamsp here.
>        List<Bitmap> bitmaps = new ArrayList<>();
> 
>        for(int i = 0; i < processedFrames.size()-1; i++){
>            ProcessedFrames processedFrame = processedFrames.get(i);
>            Bitmap bitmap = processedFrame.bitmap;
>            bitmaps.add(bitmap);
>        }
>       
> ```
> 
> Earlier I have also thought that its only possible using c/cpp or need to write jni by placing cpp codec by google. But later I have achieved to extract frames as bitamsp and also encode them back to webp aniamted image using JAVA only. I followed the WEBP container specification [docs ](https://developers.google.com/speed/webp/docs/riff_container) and modified the [b4rtaz/android-webp-encoder](https://github.com/b4rtaz/android-webp-encoder)
> 
> Also found another way with [Glide](https://github.com/bumptech/glide) . And built react native module for image transformation which requires decoding and encoding both. [react-native-image-file-transformer](https://www.npmjs.com/package/react-native-image-file-transformer)
> 
> _I have answer in case any body going through same issues about decoding encoding webp images._

Lol you private your repo, Your answer took so long i  use libwebp(c/cpp) in my android studio project and it work preety well. thank you for your response
